Formula/ykdl.rb

```diff
@@ -0,0 +1,23 @@
+class Ykdl < Formula
+  include Language::Python::Virtualenv
+
+  desc "Video downloader that focus on China mainland video sites"
+  homepage "https://github.com/zhangn1985/ykdl"
+  url "https://github.com/zhangn1985/ykdl/archive/v1.7.0.tar.gz"
+  sha256 "44e9d946c0e311a5469319a200af5bc5a6e461efa0f117f6a9608f820f22ecb8"
+  license "MIT"
+
+  depends_on "python@3.9"
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  def caveats
+    "To merge video slides, run `brew install ffmpeg`."
+  end
```

Comment:
Why not `depends_on "ffmpeg"`?

Reply:
I imitated [you-get](https://github.com/Homebrew/homebrew-core/blob/master/Formula/you-get.rb).

Reply:
What does `video slides` mean here? If it depends on ffmpeg for functionality we should define that.

Reply:
On some sites, the video was split into several parts. And `ykdl` can download all the parts then merge them into one.
And this function needs `ffmpeg`.

Reply:
Because `ffmpeg` is too big, so I do not want `ykdl` depend on it defaultly. @SMillerDev